src/components/BlogPost.js

```diff
@@ -14,12 +14,14 @@ import CommentsSection from 'components/CommentsSection';
 import Markdown from 'components/Markdown';
 import Username from 'components/Username';
 import BlogPostChangelog from 'components/BlogPostChangelog';
+import { userIsDocumentOwner, userIsDocumentCollaborator } from 'utils/Util';
+import CubePropType from 'proptypes/CubePropType';
 
-const BlogPost = ({ post, noScroll }) => {
+const BlogPost = ({ post, cube, noScroll }) => {
   const user = useContext(UserContext);
   const [editOpen, setEditOpen] = useState(false);
   const scrollStyle = noScroll ? {} : { overflow: 'auto', maxHeight: '50vh' };
-  const canEdit = user && user.id === post.owner.id;
+  const canEdit = user && (userIsDocumentOwner(user, post) || (cube && userIsDocumentOwner(cube, user)));
```

Comment:
The order of the arguments seems flipped between the two calls here.

Reply:
💯 I also don't need to the short-circuiting checks for `cube` or `user` either 
```suggestion
  const canEdit = userIsDocumentOwner(user, post) || userIsDocumentOwner(user, cube);
```